### DISCUSSION TITLE
`timedelta` field converts to `datetime` in the forms

### DISCUSSION THREAD

**Discussion by regisin** [jowilf/starlette-admin#572] [Q&A] [answered]
I have a model with a `timedelta` property. SQLAlchemy treats it correctly on the db side (it's Interval type: [Interval](https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Interval)).

However, in the admin panel, it is converted and treated as `datetime`, which is incompatible with the db. If I try to save it raises an exception.

Is there a way to override the original form?

I envision a form having several number `<input>` fields, one for each [timedelta ](https://docs.python.org/3/library/datetime.html#timedelta-objects) attribute.

Where should I start looking into?

**Comment by jowilf** [comment, marked as answer]
The interval is treated as datetime due to the `impl` attribute in the Interval type which is Datetime. 

You can create a custom field and add a converter for it. Here is how you can do it

```py
class MyCustomIntervalField(BaseField):
    ....

class MyCustomConverter(ModelConverter): # from sqla.converters
    @converts("Interval")
    def conv_interval(self, *args: Any, **kwargs: Any) -> BaseField:
        return MyCustomIntervalField(... )

admin.add_view(ModelView(MyModel, converter=MyCustomConverter()))
```

For more info:
- https://jowilf.github.io/starlette-admin/advanced/custom-field
- https://github.com/jowilf/starlette-admin/blob/main/starlette_admin/contrib/sqla/converters.py
- https://github.com/jowilf/starlette-admin/pull/131

Also feel free to contribute back, it would be good to have a field for Interval by default

**Reply by regisin** [reply]
I think I'm almost there. Last night I didn't look into the converters, so I'm putting most of the logic in the custom field class. Creating/editing a record and displaying in the list works fine (haven't done the display template yet).

Is using this converter a better way? I never noticed it was a thing..

**Reply by regisin** [reply]
Submitted #573
